frontend/src/utils/validators.js

```diff
@@ -52,62 +49,76 @@ const unique = key => withParams(
     ]
     return !includes(keys, value)
   },
-)
+))
 
-const uniqueWorkerName = withParams(
+const uniqueWorkerName = withMessage('Worker name must be unique', withParams(
   { type: 'uniqueWorkerName' },
   function unique (value) {
     return this.workers.filter(item => item.name === value).length === 1
   },
-)
+))
 
-const requiresCostObjectIfEnabled = withParams(
+const requiresCostObjectIfEnabled = withMessage('Cost Object is required', withParams(
   { type: 'requiresCostObjectIfEnabled' },
   function requiresCostObjectIfEnabled (infrastructureSecret) {
     if (!this.costObjectSettingEnabled) {
       return true
     }
     return get(infrastructureSecret, 'metadata.hasCostObject', false)
   },
-)
+))
 
-const serviceAccountKey = withParams(
-  { type: 'serviceAccountKey' },
-  function serviceAccountKey (value) {
-    try {
-      const key = JSON.parse(value)
-      if (key.project_id && alphaNumUnderscoreHyphen(key.project_id)) {
-        return true
-      }
-    } catch (err) { /* ignore error */ }
-    return false
-  },
-)
+const serviceAccountKey = withMessage('Not a valid Service Account Key',
+  withParams(
+    { type: 'serviceAccountKey' },
+    function serviceAccountKey (value) {
+      try {
+        const key = JSON.parse(value)
+        if (key.project_id && alphaNumUnderscoreHyphenPattern.test(key.project_id)) {
+          return true
+        }
+      } catch (err) { /* ignore error */ }
+      return false
+    },
+  ))
 
-const includesIfAvailable = (key, reference) => withParams(
-  { type: 'includesIfAvailable', key },
-  function includesIfAvailable (selectedKeys) {
-    const availableKeys = this[reference]
-    return includes(availableKeys, key) ? includes(selectedKeys, key) : true
-  },
-)
+const includesIfAvailable = (key, reference) => withMessage(`Value "${key}" must be selected`,
+  withParams(
+    { type: 'includesIfAvailable', key },
+    function includesIfAvailable (selectedKeys) {
+      const availableKeys = this[reference]
+      return includes(availableKeys, key) ? includes(selectedKeys, key) : true
+    },
+  ))
 
-const nilUnless = key => withParams(
-  { type: 'nilUnless', key },
-  function nilUnless (value) {
-    return !this[key] ? !req(value) : true
-  },
-)
+const nilUnless = key => withMessage(`Must not be provided if '${key}' is not set`,
+  withParams(
+    { type: 'nilUnless', key },
+    function nilUnless (value) {
+      return !this[key] ? !req(value) : true
+    },
+  ))
+
+const allWithCauserParam = (causer, validators) => {
```

Comment:
I think `causer` is not a good name. Maybe we should use collection validation and `$registerAs` to have meaningful `$propertyPath` as in this test https://github.com/vuelidate/vuelidate/blob/bdb672d334456461bcf9058dc551a21f0152d3d9/packages/test-project/src/components/CollectionValidations.vue#L9. And than map the propertyPath to a label?

Reply:
renamed to `withFieldName`